packages/artifact/src/artifact.ts

```diff
@@ -0,0 +1,102 @@
+import * as core from '@actions/core'
+import {
+  UploadSpecification,
+  getUploadSpecification
+} from './upload-specification'
+import {
+  createArtifactInFileContainer,
+  uploadArtifactToFileContainer,
+  patchArtifactSize
+} from './upload-http-client'
+import {UploadResponse} from './upload-response'
+import {UploadOptions} from './upload-options'
+import {checkArtifactName} from './utils'
+
+/**
+ * Uploads an artifact
+ *
+ * @param name the name of the artifact, required
+ * @param files a list of absolute or relative paths that denote what files should be uploaded
+ * @param rootDirectory an absolute or relative file path that denotes the root parent directory of the files being uploaded
+ * @param options extra options for customizing the upload behavior
+ * @returns single UploadInfo object
+ */
+export async function uploadArtifact(
+  name: string,
+  files: string[],
+  rootDirectory: string,
+  options?: UploadOptions
+): Promise<UploadResponse> {
+  checkArtifactName(name)
+
+  // Get specification for the files being uploaded
+  const uploadSpecification: UploadSpecification[] = getUploadSpecification(
+    name,
+    rootDirectory,
+    files
+  )
+  const uploadResponse: UploadResponse = {
+    artifactName: name,
+    artifactItems: [],
+    size: 0,
+    failedItems: []
+  }
+
+  if (uploadSpecification.length === 0) {
+    core.warning(`No files found that can be uploaded`)
```

Comment:
Should this be a warning? It seems like an error case if there are no files to upload.

Reply:
The current behavior is that and empty upload will throw a warning but not fail.

It is possible that someone includes only empty directories in the files that they want uploaded. If that is the case, the `uploadspecification` will be empty because directories get ignored by the FCS during upload.